contracts/rust/src/light_client.rs

```diff
@@ -0,0 +1,405 @@
+    /// Elapse a view with a new finalized block
+    pub fn elapse_with_block(&mut self) {
+        // if the new block is the first block of an epoch, update epoch
+        if self.state.block_height != 0
+            && self.state.block_height % self.pp.blk_per_epoch as usize == 0
+        {
+            self.epoch += 1;
+            self.st.advance();
+            self.threshold = self
+                .st
+                .total_stake(SnapshotVersion::LastEpochStart)
+                .unwrap()
+                * 2
+                / 3;
+        }
+
+        let new_root = self.new_dummy_comm();
+        let new_fee_ledger_comm = self.new_dummy_comm();
+
+        self.state.view_number += 1;
+        self.state.block_height += 1;
+        self.state.block_comm_root = new_root;
+        self.state.fee_ledger_comm = new_fee_ledger_comm;
+    }
+
+    /// Elapse a view without a new finalized block
+    /// (e.g. insufficient votes, malicious leaders or inconsecutive noterized views)
+    pub fn elapse_without_block(&mut self) {
+        self.state.view_number += 1;
+    }
+
+    /// Update stake table with `num_reg` number of new registrations and `num_exit` number of exits on L1
+    pub fn sync_stake_table(&mut self, num_reg: usize, num_exit: usize) {
+        // ensure input parameter won't exceed stake table capacity
+        let before_st_size = self.qc_keys.len();
+        assert!(self.qc_keys.len() + num_reg - num_exit <= self.pp.st_cap);
+
+        // process exits/deregister
+        for _ in 0..num_exit {
+            let exit_idx = self.rng.next_u32() as usize % self.qc_keys.len();
+            let exit_qc_key = self.qc_keys[exit_idx];
+
+            self.st
+                .deregister(&exit_qc_key)
+                .unwrap_or_else(|_| panic!("failed to deregister {}-th key", exit_idx));
+            self.qc_keys.remove(exit_idx);
+            self.state_keys.remove(exit_idx);
+        }
+
+        // process register
+        for i in 0..num_reg {
+            let bls_key: BLSVerKey = BLSOverBN254CurveSignatureScheme::key_gen(&(), &mut self.rng)
+                .unwrap()
+                .1;
+            let schnorr_key: (SchnorrSignKey, SchnorrVerKey) =
+                SchnorrSignatureScheme::key_gen(&(), &mut self.rng).unwrap();
+            let amount = U256::from(self.rng.gen_range(1..1000u32));
+
+            self.st
+                .register(bls_key, amount, schnorr_key.1.clone())
+                .unwrap_or_else(|_| panic!("failed to deregister {i}-th key"));
+            self.key_archive.insert(bls_key, schnorr_key.0.clone());
+            self.qc_keys.push(bls_key);
+            self.state_keys.push(schnorr_key);
+        }
+
+        assert!(self.qc_keys.len() == self.state_keys.len());
+        assert!(self.qc_keys.len() == before_st_size + num_reg - num_exit);
+    }
+
+    /// Elapse an epoch with `num_reg` of new registration, `num_exit` of key deregistration
+    /// Return the new light client state and a proof.
```

Comment:
This comment is confusing as the function does not return anything.

Reply:
thanks for catching it, removed in [6171539](https://github.com/EspressoSystems/espresso-sequencer/pull/854/commits/61715390c68c49dcf05f32a2505fc782957d9628)